### PR TITLE
A levels part 3 - add list of required a levels page

### PR DESCRIPTION
**Pull request by tomas-stefano**
### Context

We need to add the ability for providers to input A level requirements on TDA courses on Publish, in order to enable TDA applications.

### Changes proposed in this pull request

Screenshot 2024-06-25 at 20 43 24

Screenshot 2024-06-25 at 20 44 05

### Guidance to review

1. Does it work to add multiple A levels? The navigation between the first up to the third step.

In order to add TDA levels you need to enable rollover and see the rollover documentation

## Trello card

https://trello.com/c/3pEdlczC/1785-a-levels-3-7-add-list-of-required-a-levels-page-for-tda-courses-on-publish

### CODE REVIEW

File: app/views/publish/courses/a_level_requirements/add_a_level_to_a_list/new.html.erb [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+          </span>
+          <%= t("course.#{@wizard.current_step.model_name.i18n_key}.heading") %>
+        </h1>
+
+        <%= govuk_summary_list do |summary_list| %>
+          <% @wizard.current_step.subjects.each do |subject| %>
+            <%= summary_list.with_row do |row| %>
+              <%= row.with_key(classes: "govuk-!-font-weight-regular") { render ALevelSubjectRequirementRowComponent.new(subject) } %>
+              <%= row.with_action(text: "Change", href: "#") %>
+              <%= row.with_action(text: "Remove", href: "#") %>
```
Comment (tomas-stefano) [inline app/views/publish/courses/a_level_requirements/add_a_level_to_a_list/new.html.erb:45]: Those two links will be implemented in the next card.

File: app/wizards/a_level_steps/add_a_level_to_a_list.rb [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+    def subjects
+      Array(@subjects)
+    end
+
+    def next_step
+      if add_another_a_level == 'yes'
+        :what_a_level_is_required
+      else
+        :pending_a_level
```
Comment (tomas-stefano) [inline app/wizards/a_level_steps/add_a_level_to_a_list.rb:31]: This step doesn't exist - will be implemented into another card (this is 3 from 7). This next step is the 6th one.

File: app/controllers/publish/courses/a_level_requirements/a_level_requirements_controller.rb [hunk 3]
```diff
@@ -33,12 +34,15 @@ def create
...
+        def add_flash_message; end
+
         def verify_teacher_degree_apprenticeship_course
           redirect_to publish_provider_recruitment_cycle_courses_path(provider_code: provider.provider_code, recruitment_cycle_year: provider.recruitment_cycle_year) unless @course.teacher_degree_apprenticeship?
         end
 
         def assign_course
-          @course = CourseDecorator.new(provider.courses.find_by!(course_code: params[:course_code]))
+          @course = provider.courses.find_by!(course_code: params[:course_code])
+          @course_decorator = CourseDecorator.new(@course)
```
Comment (tomas-stefano) [inline app/controllers/publish/courses/a_level_requirements/a_level_requirements_controller.rb:45]: I have to make the distinction between the active record model and the decorator. The decorator is used in the views the active model passing to the wizard more above in this class

**Comment by tomas-stefano** [comment]
[A levels 3/7 - Add list of required A levels page for TDA courses on Publish](https://trello.com/c/3pEdlczC/1785-a-levels-3-7-add-list-of-required-a-levels-page-for-tda-courses-on-publish)

File: app/views/publish/courses/a_level_requirements/add_a_level_to_a_list/new.html.erb [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+  <% else %>
+    <%= govuk_back_link_to(
+        publish_provider_recruitment_cycle_course_a_levels_what_a_level_is_required_path(
+          @provider.provider_code,
+          @provider.recruitment_cycle_year,
+          @course.course_code
+        )
+      ) %>
+  <% end %>
+<% end %>
```
Comment (tomas-stefano) [inline app/views/publish/courses/a_level_requirements/add_a_level_to_a_list/new.html.erb:21]: I am adding/editing another card to handle another scenario when there is between 1 up to 3 A levels and the user clicks back. For now if they click "Back" is like the same as answering yes and anothing another A level (up to 4 maximum). This is the card https://trello.com/c/mEPXi2RG/1868-a-levels-edge-case-back-link-when-user-is-adding-a-levels?filter=label:Epic%3A%20TDA (I am still editing as I wrote this comment)

File: config/locales/en.yml [hunk 5]
```diff
@@ -276,10 +276,15 @@ en:
         about_accrediting_provider:
           text: "Enter details about the accredited provider"
   value_not_entered: "Not entered"
+  a_level_grades:
+    minimum_grade: "Grade %{minimum_grade} or above"
```
Comment (elceebee) [inline config/locales/en.yml:280]:
```suggestion
    minimum_grade: Grade %{minimum_grade} or above
```

Not a big deal, but you only need this with funny characters (like the max_grade below). The interpolation doesn't need quotes.

File: config/locales/en.yml [hunk 6]
```diff
@@ -292,6 +297,10 @@ en:
           any_humanities_subject: Any humanities subject
           any_science_subject: Any science subject
           other_subject: Choose a subject
+      add_a_level_to_a_list:
+        add_another_a_level_options:
+          "yes": "Yes"
+          "no": "No"
```
Comment (elceebee) [inline config/locales/en.yml:303]:
```suggestion
          yes: Yes
          no: No
```

**Review by elceebee — APPROVED** [review]
Tested locally and everything is as expected. 🚀

File: config/locales/en.yml (hunk 5 above)
Comment (tomas-stefano) [reply config/locales/en.yml:280]: True. I am so confused in the locale. Some has quotes other doesn't.

File: config/locales/en.yml (hunk 6 above)
Comment (tomas-stefano) [reply config/locales/en.yml:303]: actually, if I remove the quotes it won't work as YAML will parse as true or false.

**Pull request merged**